src/index.js

```diff
@@ -24,77 +24,98 @@ function App() {
 
     useEffect(() => {
         const storedTodos = JSON.parse(localStorage.getItem(LOCAL_STORAGE_KEY_DATA));
-        setToDos(storedTodos)
+        setToDos(storedTodos);
     }, []); // run and clean effect only once on mount and unmount, this effect doesnt depend on any props or state so it never re-runs
     // we only call this retrieving from localStorage once, because we only need to load from saved items when page first loads up, never again
 
     useEffect(() => {
-        localStorage.setItem(LOCAL_STORAGE_KEY_DATA, JSON.stringify(todos))
-    },[todos]); // this effect only fires if the todos array changes
+        localStorage.setItem(LOCAL_STORAGE_KEY_DATA, JSON.stringify(todos));
+    }, [todos]); // this effect only fires if the todos array changes: which is toggle complete and remove tasks
 
-    function handleAddItem(e) {
-        const currentInput =  toDoInput.current.value;
+    const handleAddItem = () => {
+        const currentInput = toDoInput.current.value;
         const pickedDueDate = dueDateInput.current.value;
 
-        if(currentInput && pickedDueDate) {
-            setToDos(prevList => {
-                return [...prevList, { id:uuidv4(), name: currentInput, due: pickedDueDate, completed: false}]
-            })
+        if (currentInput && pickedDueDate){ // will implement validations in next step
+            const newTodoItem = {
+                id: uuidv4(),
+                name: currentInput,
+                due: pickedDueDate,
+                completed: false
+            };
+
+            setToDos([...todos, newTodoItem]);
+            toDoInput.current.value = null;
+            dueDateInput.current.value = null;
         }
-        toDoInput.current.value = null;
-        dueDateInput.current.value = null;
     }
 
-    function toggleCompleted(e) {
-        const currentTodos = [...todos];
-        const toToggleItem = currentTodos.find(eachCurrentTodo => eachCurrentTodo.id === e);
+    const toggleCompleted = (taskID) => {
+        const toToggleItem = todos.find(({ id }) => id === taskID);
```

Comment:
**Naming** 
The item is already toggled.
```suggestion
        const toggledItem = todos.find(({ id }) => id === taskID);
```

Reply:
`toggleCompleted` as this function's name is probably misleading since we are about to change the completion status, either to be completed or uncompleted. I have renamed this function to be `toggleCompletion`.

I have came up with a naming system for the DOM events:
verb (action on the DOM) + verb (manipulation on the DOM/data) + noun. So:

toggleSort('task') --> toggleSortColumn('task')
handleAddItem
handleClearItems
toggleCompletedClick --> toggleCompleteItem(id)